app/firebase/functions/src/platforms/twitter/twitter.service.ts

```diff
@@ -273,18 +273,30 @@ export class TwitterService
       throw new Error('Unexpected undefined refresh token');
     }
 
+    const credentials: TwitterUserCredentials = {
+      accessToken: result.accessToken,
+      refreshToken: result.refreshToken,
+      expiresIn: result.expiresIn,
+      expiresAtMs: this.time.now() + result.expiresIn * 1000,
+    };
+
     const twitter: TwitterUserDetails = {
       user_id: user.id,
       signupDate: 0,
-      write: {
-        accessToken: result.accessToken,
-        refreshToken: result.refreshToken,
-        expiresIn: result.expiresIn,
-        expiresAtMs: this.time.now() + result.expiresIn * 1000,
-      },
       profile: user,
     };
 
+    /** the same credentials apply for reading and writing */
+    if (data.type === 'write') {
+      twitter[data.type] = credentials;
+    }
+
+    /** always store the credential as read credentials */
+    twitter.read = {
```

Comment:
This mirrors the write tokens into the read access tokens, which means they will be the same and will need to be refreshed together, whether the refresh is triggered from a read or write action.

I see two ways to handle this:
- in the twitter service, whenever refreshing the tokens, if refreshing the read tokens, check if write tokens exist, and refresh those as well, and vice versa.
- don't always store the write credentials as read credentials as well and instead assume read credentials already exist, thereby handling read and write tokens completely separately. This would mean in our tests having to grant read and write tokens separately.

Reply:
I opted for the first option in the next commit, since it is simpler. Though we will probably want to handle the tokens separately.